doc/lua_api.txt

```diff
@@ -617,6 +640,64 @@ Result is more like what you'd expect if you put a color on top of another
 color, meaning white surfaces get a lot of your new color while black parts
 don't change very much.
 
+A Multiply blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:-64^[overlay:textureB.png
+
+#### `[screen:<color>`
+
+Apply a Screen blend with the given color. A Screen blend is the inverse of
+a Multiply blend, lightening images instead of darkening them.
+
+`<color>` is specified as a `ColorString`.
+
+A Screen blend can be applied between two textures by using the overlay 
+modifier with a brightness adjustment:
+
+    textureA.png^[contrast:0:64^[overlay:textureB.png
+
+#### `[hsl:<hue>:<saturation>:<lightness>`
+
+Adjust the hue, saturation, and lightness of the texture. Like
+"Hue-Saturation" in GIMP, but with 0 as the mid-point.
+
+`<hue>` should be from -180 to +180
+
+`<saturation>` and `<lightness>` are optional, and both from -100 to +100
+
+#### `[contrast:<contrast>:<brightness>`
+
+Adjust the brightness and contrast of the texture. Conceptually like
+GIMP's "Brightness-Contrast" feature but allows brightness to be wound 
+all the way up to white or down to black.
+
+`<contrast>` and `<brightness>` are both values from -127 to +127.
+
+`<brightness>` is optional.
+
+#### `[overlay:<file>`
+
+Applies an Overlay blend with the two textures, like the Overlay layer mode 
+in GIMP. Overlay is the same as Hard light but with the role of the two 
+textures swapped, see the `[hardlight` modifier description for more detail
+about these blend modes.
+
+#### `[hardlight:<file>`
+
+Applies a Hard light blend with the two textures, like the Hard light layer 
+mode in GIMP. 
+
+Hard light combines Multiply and Screen blend modes. Light parts of the 
+`<file>` texture will lighten (screen) the base texture, and dark parts of the
+`<file>` texture will darken (multiply) the base texture. This can be useful
+for applying embossing or chiselled effects to textures. A Hard light with the
+same texture looks like an S-curve.
```

Comment:
What does S-curve mean? I don't see any difference.
![grafik](https://user-images.githubusercontent.com/1497498/132958744-19339d11-c388-4e5c-9df7-a0c97a512569.png)

Left: `testimage.png^[hardlight:testimage.png`
Right: `testimage.png`

Reply:
I'll have to investigate, they should look noticeably different.

[This page show an s-curve on an image](https://www.lightxapp.com/curves-tool-in-photo-editing/): the lights should get lighter and the darks should get darker

Reply:
It works, phew! It's just hard to see in that test image.

I've added some arrows where the lights are noticeably lighter and the darks are noticeably darker, and demo'd it with the image that webpage used.
![hardlight screenshot](https://user-images.githubusercontent.com/6390507/132973744-200e1fdc-aa3d-4463-9e46-5cb94ab9a59e.jpg)

Because this is a better way to increase contrast than with the contrast modifier, I think it could be added as a note in the contrast modifier's documentation.

Reply:
I've added the note to the `[contrast` documentation in [fb393c1](https://github.com/minetest/minetest/pull/10100/commits/fb393c1107f2e8d713943f3fbd3194da9bc21415#diff-da915d7065166fede89169efbfe67ec7016a1458fa6a553443039f2f1ef7a779R690-R693)

and also hopefully made it less cryptic in the hard light [desciption](https://github.com/minetest/minetest/pull/10100/commits/fb393c1107f2e8d713943f3fbd3194da9bc21415#diff-da915d7065166fede89169efbfe67ec7016a1458fa6a553443039f2f1ef7a779R712-R714)